crawl-ref/source/melee-attack.cc

```diff
@@ -689,6 +689,7 @@ bool melee_attack::handle_phase_killed()
 {
     if (attacker->is_player()
        && ( you.species == SP_DUSK_WALKER
+            && x_chance_in_y(1,3)
             || you.form == transformation::hydra)
```

Comment:
I would put the `you.species == SP_DUST_WALKER && one_chance_in(3)` check on one line, and use `one_chance_in(3)` rather than `x_chance_in_y(1, 3)` because it's neater.